gprofiler/merge.py

```diff
@@ -332,6 +333,41 @@ def _enrich_and_finalize_stack(
     return f"{enrich_data.application_prefix}{enrich_data.container_prefix}{stack} {count}"
 
 
+def concatenate_from_external_file(
+    collapsed_file_path: str,
+    obtained_metadata: Metadata,
+) -> Tuple[str, str, str]:
+    """
+    Concatenate all stacks from all stack mappings in process_profiles.
+    Add "profile metadata" and metrics as the first line of the resulting collapsed file.
+    """
+
+    lines = []
+    start_time = None
+    end_time = None
+
+    # TODO: container names and application metadata
+    with open(collapsed_file_path) as file:
+        for index, line in enumerate(file):
+            if index == 0:
```

Comment:
Ensure the line starts with `#` as the format expects. (can be `assert line.startswith(...)`)